lua/telescope/previewers.lua

```diff
@@ -443,42 +541,102 @@ previewers.vimgrep = defaulter(function(opts)
   }
 end, {})
 
-previewers.ctags = defaulter(function(_)
+previewers.get_file_previewer = function(opts)
+  if config.values.vim_buffers_everywhere then
+    return previewers.file.new(opts)
+  else
+    return previewers.cat.new(opts)
+  end
+end
+
+previewers.get_vimgrep_previewer = function(opts)
+  if config.values.vim_buffers_everywhere then
+    return previewers.internal_vimgrep.new(opts)
+  else
+    return previewers.vimgrep.new(opts)
+  end
+end
+
+previewers.get_qflist_previewer = function(opts)
+  if config.values.vim_buffers_everywhere then
+    return previewers.internal_vimgrep.new(opts)
+  else
+    return previewers.qflist.new(opts)
+  end
+end
+
+previewers.file = defaulter(function(_)
+  return previewers.new_buffer_previewer {
+    define_preview = function(self, entry, status)
+      with_preview_window(status, nil, function()
+        local path = from_entry.path(entry, true)
+        if path == nil then return end
+        file_maker(path, self.state.bufnr)
+      end)
+    end
+  }
+end, {})
+
+previewers.internal_vimgrep = defaulter(function(_)
   return previewers.new_buffer_previewer {
     setup = function()
-      return {}
+      return {
+        last_set_bufnr = nil
+      }
     end,
 
+    teardown = function(self)
+      if self.state and self.state.hl_id then
+        vim.api.nvim_buf_clear_namespace(self.state.last_set_bufnr, previewer_ns, 0, -1)
+      end
+    end,
+
+    define_preview = function(self, entry, status)
+      with_preview_window(status, nil, function()
+        vim.api.nvim_buf_clear_namespace(self.state.last_set_bufnr, previewer_ns, 0, -1)
+
+        local lnum = entry.lnum or 0
+        local path = from_entry.path(entry, true)
+        if path == nil then return end
+
+        file_maker(path, self.state.bufnr, function()
+          -- TODO(conni2461): We are sometimes outside of the buffer. Especially if we scroll through the list fast. hmm
+          if entry.lnum then
+            vim.api.nvim_buf_add_highlight(self.state.bufnr, self.state.winid, "Visual", entry.lnum - 1, 0, -1)
+            vim.api.nvim_win_set_cursor(status.preview_win, {lnum, 0})
+          end
+
+          self.state.last_set_bufnr = bufnr
+        end)
```

Comment:
@tjdevries 
[file_maker](https://github.com/nvim-telescope/telescope.nvim/pull/298/files#diff-9d44c37e482d0e8fff742a54123a2a369c4788ed803b22d9ebc5da95b46564d1R67-R85) here defined.

This currently does not work because not all lines are set yet. I though i would fix it if i pass that code as callback and run it after `vim.api.nvim_buf_set_lines` but that did not help. Am i missing something?